e2e/test_payment.py

```diff
@@ -197,12 +197,15 @@ def verified_seat_payment_with_credit_card(self, selenium, addresses):
 
                 # We finished a test, stop trying course runs
                 break
-            except TimeoutException as exc:
+            except TimeoutException:
+                log.warning("Failed to get a valid course run for SKU %s, continuing", verified_seat['sku'])
```

Comment:
Does this make sense?

```suggestion
                log.warning("TimeoutException while getting searching for course run for SKU %s, continuing", verified_seat['sku'])
```

Reply:
Idea fairy:

An alternative is:

```
if "No product is available" in exc.msg:
    (existing log message)
    continue
else:  # TODO: Remove else clause after investigation
    (new log message)
    continue
raise
```

In case it would be useful to separate out "no product is available" errors from the rest of the TimeoutExceptions.

May be worth logging `exc.msg` in the `(new log message)` as well.

Reply:
Yes I like this better! Let's go one further: 'while trying to add course run to basket for' (It will only let me 'commit suggestion', not edit it)